### PR TITLE
[generatePin] Avoid generating weak PINs

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview
Closes https://github.com/votingworks/vxsuite/issues/2525:
- Creates denylist of common weak pin patterns (repeated digits, sequential digits) and checks against them when generating random PINs.
-  Retries random PIN generation until we find a non-weak PIN.
- Also raises minimum PIN length from 1 to 6
  - Tangentially: there's probably room for eliminating the option of specifying PIN length altogether, unless we plan on having PINs of different lengths used in different parts of the system at some point)

## Testing Plan 
- Unit tests now mock `randomBytes` to allow for determinism - we mock a few weak PIN return values and then a non-weak one and verify that `generatePin` skips over the weak ones.
- Also some manual verification locally with "all-zero" feature flag on and off

## Checklist
- [n/a] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [x] I have added JSDoc comments to any newly introduced exports

- [n/a] I have added a screenshot and/or video to this PR to demo the change
- [n/a] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Review by eventualbuddha — COMMENTED** [review]
Code LGTM. I might also want one test that unmocks `randombytes` and uses `fast-check` to verify the property "no PINs are weak". It's easiest to do so as a separate test file due to the way jest mocks work:

```ts
// pins.prop.test.ts
import fc from 'fast-check';
import { generatePin, isWeakPin, MIN_PIN_LENGTH } from './pins';

test('no PINs are weak', () => {
  fc.assert(
    fc.property(fc.integer(MIN_PIN_LENGTH, 100), (length) => {
      const pin = generatePin(length);
      expect(isWeakPin(pin)).toBe(false);
    })
  );
});
```

I want to leave final approval for @benadida or @umbernhard for security review.

File: libs/utils/src/pins.ts
Comment (arsalansufi) [inline libs/utils/src/pins.ts:44]: > Tangentially: there's probably room for eliminating the option of specifying PIN length altogether, unless we plan on having PINs of different lengths used in different parts of the system at some point

Agreed! I think it's unlikely that we'll allow users to pick a PIN length other than 6, and I like the idea of simplifying

**Review by arsalansufi — APPROVED** [review]
LGTM!

**Comment by kofi-q** [comment]
> I might also want one test that unmocks `randombytes` and uses `fast-check` to verify the property "no PINs are weak".

ah makes sense! i might set the upper limit somewhere between 9 and 16 (credit card # length?) just to be safe - worried longer PINs are more likely to contain one these "weak PIN" substrings and the retry loop could spin for a while (and also since we won't expect users to memorise anything much longer).
i considered (and punted on) making the substring patterns variable length, depending on the PIN length, which might be necessary if we expect to be generating longer PINs for other use cases

File: libs/utils/src/pins.real_random.test.ts
Comment (benadida) [reply libs/utils/src/pins.real_random.test.ts:12]: seeing some of the tests below that ensure that the weak PINs are not returned is also a good alternative to what I proposed. Either way, I think this test here is not particularly useful and should probably be removed.

**Review by benadida — APPROVED** [review]

File: libs/utils/src/pins.real_random.test.ts
Comment (eventualbuddha) [reply libs/utils/src/pins.real_random.test.ts:12]: My experience with `fast-check` tells me that the failures tend to either happen almost every time or not at all, though this does depend on the search space and how sparsely populated it is with PINs returned by `generatePin` but flagged by `isWeakPin`. This test mostly just ensures that these functions are kept in sync and comes up with random test values. If you prefer to hardcode those values that's fine as long as you're quite confident you've covered the search space well.

Comment (kofi-q) [reply libs/utils/src/pins.real_random.test.ts:12]: Hm yeah, also in favour of dropping this - it did seem like there's the possibility of false positives, especially when I tested it with a lower MAX_PIN_LENGTH earlier. It also relies on an assumption that `isWeakPin` does catch all possible weak PINs that we care about. Going to instead add a few more obvious weak PIN samples to the other test file to give us more coverage.

**Pull request merged**
